scripts/distro_codegen.py

```diff
@@ -107,6 +107,23 @@ def collect_template_dependencies(template_dir: Path) -> tuple[str | None, list[
     return None, []
 
 
+def generate_dependencies_files(change_tracker: ChangedPathTracker):
```

Comment:
(No action required, Would require significant refactoring) At this point, why do we even need to have `pip_packages` in code? Can't these be directly managed in the respective `requirements.txt` files? (Or even directly in `pyproject.toml`?) I always found tracking python dependencies in python code somewhat distasteful; do you think it's a worthy path?